viper.py

```diff
@@ -510,12 +517,28 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
             w_ok = w[i_ok]
             f_ok = f[i_ok]
 
-            if tplname:
-                p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s,t, c=cc, c0=c0, dx=0.1, sig=sig[i_ok])
+            if tplname or createtpl:
+                p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s,t, c=cc, c0=c0, dx=0.1, sig=sig[i_ok], res=res, rel_fac=rel_fac)
             else:
             # do not fit for velocity
                 p, e_p = S_mod.fit(x_ok, f_ok, a=a, b=bg, s=s,t=t0, c=cc, v0=0, c0=c0, dx=0.1, sig=sig[i_ok])
 
+    if createtpl:
+	# save telluric corrected spectrum
+
+        S_tell = S_mod(x, *p)		# modeled telluric spectrum
+        S_tell /= np.nanmean(S_tell)	
+
+        tf = f / S_tell			# telluric corrected spectrum
+        tf[S_tell<0.2] = np.nan
+        w1 = np.poly1d(p[2][::-1])(x-icen)
+
+        tpl_all[o,0][n] = w1   
+        tpl_all[o,1][n] = np.interp(w1 / (1+berv/c), w1, tf / np.nanmedian(tf))
+
+        tpl_all[o,2][n] = S_tell * 1./(e/np.nanmean(f))   #**2
```

Comment:
The comment this line with `downweighting by telluric spectrum`.

Reply:
Why flux and error have different scaling factors `nanmedian(tf)` vs. `nanmean(f)`?

Reply:
I didn't thought too much about that. I just used `nanmedian(tf)` beacuse I wanted to scale `tf` and `nanmean(f)` because `e` is related to the non-corrected spectrum `f`. Don't know if it would make much of a difference.

Reply:
I find the error propagation/manipulation a bit unclear. Each operation one flux should also be done on uncertainties.
I'm suggesting
```
    tf = f / S_tell			# telluric corrected spectrum
    e_tf = e / S_tell
    norm = np.nanmedian(tf)
    tf /= norm
    e_tf /= norm
    ...
    tpl_all[o,2][n] = S_tell / e_tf**2   # weights (downweighted by telluric spectrum)
```
Note for relative weights:
current version: `wgt = S_tell / e_tf = S_tell / e`
my version: `wgt = S_tell / e_tf**2 = S_tell / (e / S_tell)**2 = S_tell**3 / e**2`
For photon noise S = e^2 and a star with no lines S_tell = S = e^2:
current version: `wgt -> e = S_tell**0.5`
my version: `wgt -> e**6 / e**2 = e**4 = S_tell**2`
Hm, the exponent basically differs by 4 (!). An exponent of two (instead of 0.5) would downweight telluric absorption much stronger. (Your version with e**2 instead of e would have exponent 1.)

Reply:
I get your point and started doing tests with both weightings (yours and mine). In the moment it is hard to judge which template gives the better results. In some cases the obtained RVs look better with the template created with your weighting, in some cases it looks better with my way of weighting. I have to look into it in more detail and go on with tests on other targets. But it takes a bit of time (especially as the number of data is pretty large). I will come back to you, when I know more.